src/outputs/restart.cpp

```diff
@@ -17,30 +17,582 @@
+
+  // Mesh information
+  localDSpace = H5Screate(H5S_SCALAR);
+  myDSet = H5Dcreate(file, "/Mesh", PREDINT32, localDSpace, H5P_DEFAULT, H5P_DEFAULT,
+                     H5P_DEFAULT);
+
+  auto &nx1 = pmb->block_size.nx1;
+  auto &nx2 = pmb->block_size.nx2;
+  auto &nx3 = pmb->block_size.nx3;
+  int bsize[3] = {nx1, nx2, nx3};
+  nLen = 3;
+  localnDSpace = H5Screate_simple(1, &nLen, NULL);
+  status = writeH5AI32("blockSize", bsize, file, localnDSpace, myDSet);
+  status = H5Sclose(localnDSpace);
+
+  status = writeH5AI32("nbtotal", &pm->nbtotal, file, localDSpace, myDSet);
+  status = writeH5AI32("nbnew", &pm->nbnew, file, localDSpace, myDSet);
+  status = writeH5AI32("nbdel", &pm->nbdel, file, localDSpace, myDSet);
+  status = writeH5AI32("rootLevel", &rootLevel, file, localDSpace, myDSet);
+  status = writeH5AI32("MaxLevel", &max_level, file, localDSpace, myDSet);
+
+  { // refinement flag
+    int refine = (pm->adaptive ? 1 : 0);
+    status = writeH5AI32("refine", &refine, file, localDSpace, myDSet);
+
+    int multilevel = (pm->multilevel ? 1 : 0);
+    status = writeH5AI32("multilevel", &multilevel, file, localDSpace, myDSet);
+  }
+
+  { // mesh bounds
+    const auto &rs = pm->mesh_size;
+    const Real limits[6] = {rs.x1min, rs.x2min, rs.x3min, rs.x1max, rs.x2max, rs.x3max};
+    const Real ratios[3] = {rs.x1rat, rs.x2rat, rs.x3rat};
+    nLen = 6;
+    localnDSpace = H5Screate_simple(1, &nLen, NULL);
+    status = writeH5AF64("bounds", limits, file, localnDSpace, myDSet);
+    status = H5Sclose(localnDSpace);
+
+    nLen = 3;
+    localnDSpace = H5Screate_simple(1, &nLen, NULL);
+    status = writeH5AF64("ratios", ratios, file, localnDSpace, myDSet);
+    status = H5Sclose(localnDSpace);
+  }
+
+  { // boundary conditions
+    nLen = 6;
+    localnDSpace = H5Screate_simple(1, &nLen, NULL);
+    int bcsi[6];
+    for (int ib = 0; ib < 6; ib++) {
+      bcsi[ib] = static_cast<int>(pm->mesh_bcs[ib]);
+    }
+    status = writeH5AI32("bc", bcsi, file, localnDSpace, myDSet);
+    status = H5Sclose(localnDSpace);
+  }
+
+  // close space and set
+  status = H5Sclose(localDSpace);
+  status = H5Dclose(myDSet);
+
+  // end mesh section
+
+  // write blocks
+  // MeshBlock information
+  // Write mesh coordinates to file
+  hsize_t local_start[5], global_count[5], local_count[5];
+  hid_t gLocations;
+
+  local_start[0] = 0;
+  local_start[1] = 0;
+  local_start[2] = 0;
+  local_start[3] = 0;
+  local_start[4] = 0;
+  for (int i = 0; i < Globals::my_rank; i++) {
+    local_start[0] += nblist[i];
+  }
+  hid_t property_list = H5Pcreate(H5P_DATASET_XFER);
+#ifdef MPI_PARALLEL
+  H5Pset_dxpl_mpio(property_list, H5FD_MPIO_COLLECTIVE);
+#endif
+
+  // set starting poing in hyperslab for our blocks and
+  // number of blocks on our PE
+
+  // open blocks tab
+  hid_t gBlocks = H5Gcreate(file, "/Blocks", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+
+  // write Xmin[ndim] for blocks
+  {
+    Real *tmpData = new Real[num_blocks_local * 3];
```

Comment:
I believe this is spelled `std::vector<Real> tmpData(num_blocks_local*3);`

Reply:
And you would be wrong.  That would only be for single-lingual programmers.  For bilingual programmers using the C interface of HDF5, this is actually spelled `void *tmpData;`

Reply:
As I mentioned in another comment, I argue for pulling out the pointer from a vector and then passing it in to HDF5. I don't like the new and deletes floating around.

Reply:
This is serious enough that I should fix it now.

Reply:
Done.